File: 190_Reverse_Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t rever = 0;
        uint32_t m = n;
        for(int i=0; i<32; i++){
            int mod = m%2;
            rever = rever<<1;
            rever = rever+mod;
            m/=2;
        }
        return rever;
    }
};

File: 190_Reverse_Bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int rever = 0;
        int m = n;
        for(int i=0; i<32; i++){
            int mod = m&1;
            rever = rever<<1;
            rever = rever+mod;
            m>>>=1;
        }
        return rever;
    }
}

